File: 0048-rotate-image/0048-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        for r in range(len(matrix)//2):
            reverse = len(matrix)-1-r
            matrix[r], matrix[reverse] = matrix[reverse], matrix[r]

        for r in range(len(matrix)):
            for c in range(r+1,len(matrix[0])):
                matrix[r][c], matrix[c][r] = matrix[c][r], matrix[r][c]
        
        

        

File: 0048-rotate-image/0048-rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& matrix) {
        for (int r=0;r<matrix.size()/2; r++){
            int reverse = matrix.size()-1-r;
            swap(matrix[r],matrix[reverse]);
        }

        for (int r=0;r<matrix.size(); r++){
            for (int c=r+1;c<matrix[0].size(); c++){
                swap(matrix[r][c],matrix[c][r]);
            }
        }
    }
};